src/serialization/sb3.js

```diff
@@ -288,17 +315,18 @@ const getExtensionIdForOpcode = function (opcode) {
 /**
  * Serialize the given blocks object (representing all the blocks for the target
  * currently being serialized.)
- * @param {object} blocks The blocks to be serialized
+ * @param {object} target The target containing the blocks to be serialized.
  * @return {Array} An array of the serialized blocks with compressed inputs and
  * compressed primitives and the list of all extension IDs present
  * in the serialized blocks.
  */
-const serializeBlocks = function (blocks) {
+const serializeBlocks = function (target) {
```

Comment:
The problem I ran into is that variable serialization wants to call `target.lookupVariableByNameAndType(...)` in order to get the variable's name and type, so I needed to feed `target` all the way down the line. That's a bit awkward since the "target" being passed into `serializeBlocks` here is the flattened target, not the "real" target, so variable serialization does `target = this.runtime.getTargetById(target.id);` to get the real target before calling `target.lookupVariableByNameAndType`.

Given that, I started thinking we should just pass down the target ID instead of the flattened target object. Then I started thinking about passing a "context" object in case we need more context info in the future instead of just the target ID. Then I was like, "well, the flattened target object is kind of the whole context" -- which is how I got here. Maybe it would be better to just pass in the target ID instead, if for no other reason than to avoid hassling extension authors with flattened targets.

Reply:
Agreed about hassling extension authors with flattened targets. I'm okay with passing down a target ID but that also assumes that everywhere the target will be used (currently only in the custom serialization code) there will be access to the runtime. Maybe we can just pass around the hydrated target and call it `runtimeTarget` or something to indicate that it's not flattened? 

Related to a different suggestion I made however, it seems like we might also need to start passing around the `runtime` as well, in which case `runtime + targetID` is okay too.